lib-ee/emqx_ee_schema_registry/README.md

```diff
@@ -0,0 +1,69 @@
+# EMQX Schema Registry
+
+EMQX Schema Registry for managing various of schemas for
+decoding/encoding messages.
+
+To use schema in rule engine, a schema name should be passed to the
+SQL functions that decode/encode data, like:
+
+```sql
+SELECT
+     schema_decode('sensor_notify', payload) as payload
+FROM
+     "message.publish"
+WHERE
+     topic = 't/1'
+```
+
+## Using schema registry with rule engine
+
+```
+                      +---------------------------+
+                     |                           |
+ Events/Msgs         |                           |   Events/Msgs
+ -------------------->           EMQX            |------------------>
+                     |                           |
+                     |                           |
+                     +-------------|-------------+
+                                   |
+                             HOOK  |
+                                   |
+                     +-------------v-------------+           +----------+
+                     |                           |   Data    |          |
+                     |        Rule Engine        ------------- Backends |
+                     |                           |           |          |
+                     +------|-------------|------+           +----------+
+                            |^            |^
+                      Decode||            ||Encode
+                            ||            ||
+                     +------v|------------v|-----+
+                     |                           |
+                     |      Schema Registry      |
+                     |                           |
+                     +---------------------------+
+```
+
+## Architecture
+
+```
+                              |              |
+                       Decode |    [APIs]    | Encode
+                              |              |
+                              |              |           [Registry]
+                       +------v--------------v------+
+    REGISTER SCHEMA    |                            |
```

Comment:
Good points, thanks!  I've added a comment clarifying "register schema" and removed the "load parsers" flow.

We only have Avro in this PR to avoid making it too big.  Protobuf and others will come in subsequent PRs.